src/components/common/UserProfileEdit/UserAdvancedSettings.tsx

```diff
@@ -123,7 +123,16 @@ const UserAdvancedSettings = ({ user: { walletAddress, profile } }: Props) => {
               <AdvancedSettingsRow
                 name={row.name}
                 paragraphText={row.paragraphText}
-                toggleDisabled={!metatransactionsAvailable}
+                toggleDisabled={(() => {
+                  if (row.name === 'metatransactionsEnabled') {
+                    return !metatransactionsAvailable;
+                  }
+                  // Not implemented currently, so disable it outright
+                  if (row.name === 'decentralizedModeEnabled') {
+                    return true;
+                  }
+                  return true;
+                })()}
```

Comment:
I think this logic should be moved up to the `getAdvancedSettingsRows` fn. In there you can add a `disabled` field that's `true` for `decentralizedModeEnabled` and `!metatransactionsAvailable` for `metatransactionsEnabled`